### ISSUE TITLE
Add support for Component decorator host property.

### ISSUE THREAD

**Issue by x-strong** [angular/vscode-ng-language-service#1730]
**Is your feature request related to a problem? Please describe.**

# 🚀 feature request

### Description

Add support for the host configuration item of Component decorator:
```ts

1. @Component({
2.   host: {
3.    '[class.dragging]': 'dragging',
4.   }
5. })
6. class DraggableComponent {
7.    public dragging = true;
8. }

```

### Feature Type
What does this bug affect

- [x] Angular Language Service VSCode extension
- [x] Angular Language Service server

### Describe the solution you'd like
 
Take the code given before as an example, the 'dragging' property in line 3 should can be auto completed, highlighted, navigate to, rename, and show comment popup when mouse hover.

Further more, description specific to key in the pattern supported by Angular (for example, '[class.dragging]' means add CSS class dragging to the host element when the value (here, is the dragging property of the component) evaluates to true, and removed when evaluates to false) of host should be popup as common document.

### Anything else relevant?

Reference the JetBrains WebStorm Dev Experience.

**Comment by atscott** [comment]
Yep, I think this is a good idea and really shouldn’t be too hard to implement (famous last words). But we have all the pieces in other places - they just need to be put together in the host meta as well.

**Comment by dasco144** [comment]
Is there a plan to add this support in the future? With the docs now [recommending this approach](https://angular.dev/guide/components/host-elements#the-hostbinding-and-hostlistener-decorators) for host binding (over the decorator), this feels like a bit of a gap.
```
Always prefer using the host property over [@HostBinding](https://angular.dev/api/core/HostBinding) and [@HostListener](https://angular.dev/api/core/HostListener). These decorators exist exclusively for backwards compatibility.
```

**Comment by zwarag** [comment]
Looks like there is a contradiction in the docs.

> Consider preferring the @HostListener and @HostBinding to the host property of the @Directive and @Component decorators.

https://angular.dev/style-guide#style-06-03

**Comment by danny-huang-openfind** [comment]
Looks like this will be fixed in [angular/angular#52561](https://github.com/angular/angular/issues/52561), so again, will this feature release in the future? cause this can be an urgent need someday.

**Comment by SeraphyBR** [comment]
👀
